instrumentation/opentelemetry-instrumentation-grpc/src/opentelemetry/instrumentation/grpc/_server.py

```diff
@@ -71,6 +71,30 @@ def __init__(self, servicer_context, active_span):
         self.details = None
         super().__init__()
 
+    @property
```

Comment:
Still, we should not use someone else's private attributes. Also, next time we need to access a private attribute we would need to add yet another property.

I feel like this is a job for [`__getattr__`](https://docs.python.org/3/reference/datamodel.html#object.__getattr__).

Reply:
I could agree with that, though since this class is a wrapper instead of a subclass, this is always going to be a problem when the wrapped class gets changed and people expect those changes to propagate.  In fact, now that I look again, we're missing a private function as well, `_finalize_state()`.

Not sure who the original author was, but I wonder why the choice was made to wrap this rather than subclass it?  Possibly because the `grpc` implementation is also marked private.

What would your suggestion be, to redefine `__getattr__` so it calls `getattr()` on the wrapped class if the attribute is not defined in our wrapper?

Reply:
Correct, use the wrapped attribute if the attribute is not found in the wrapper. :+1:

Reply:
Oh I know the answer to my question - because the base class is a metadata class, the real class is that private `_Context` in the private `grpc._server` module.

Expect an update later today, and thanks for the tip!

Reply:
:v: